account_banking_payment_transfer/model/account_payment.py

```diff
@@ -154,10 +154,12 @@ def _prepare_move_line_transfer_account(
             partner_id = payment_lines[0].partner_id.id
             name = _('%s line %s') % (
                 labels[self.payment_order_type], payment_lines[0].name)
+            date_maturity = payment_lines[0].date
         else:
             partner_id = False
             name = '%s %s' % (
                 labels[self.payment_order_type], self.reference)
+            date_maturity = False
```

Comment:
I think you can use payment_lines[0].date in this part too, because lines will be grouped by execution date.